Blockbase: refactor footers to patterns

MaggieCabrera opened the pull request:
<!-- Thanks for contributing to our free themes! Please provide as much information as possible with your Pull Request by filling out the following - this helps make reviewing much quicker! -->

#### Changes proposed in this Pull Request:

This is a WIP

This PR repeats the work done for headers in https://github.com/Automattic/themes/pull/5238 but for footers. We are removing duplicate code and giving the user options to use all possible footers designed between our themes no matter the theme you installed.

I have a couple of comments I'd like to discuss:

- I've not wrapped the "Proudly Powered by Wordpress" texts for translation because we have the plugin on dotcom to do that for us. I think it would be safe to do it anyway, let me know what you think.
- Should we make the pattern from https://github.com/Automattic/themes/issues/5025 be the default footer for Geologist and Zoologist even though those are already launched? Or should we just keep them as a pattern that users can swap to if they want?
- Skatepark has some text under the title: 

> Skatepark's coaches will work with you to develop and improve your skating abilities.

Should we change this text to something generic that works for all themes? Or maybe we should make this be the tagline block instead? @kjellr @beafialho  I'd love your input on this

Seedlet blocks footer was very different from the original, I tried to make it closer to what it should be. Also ended up opening https://github.com/WordPress/gutenberg/pull/37586 because it's related, I think it's fine if that doesn't get merged though.

![Screen Capture on 2021-12-22 at 13-20-48](https://user-images.githubusercontent.com/3593343/147092034-2a575ffe-b8be-4137-8840-a893f3e35343.gif)

The new pattern on geologist with the small search block block style (I didn't apply it to the theme, just made the screenshot, this pattern is not used in any theme at the moment in this PR, pending discussion):

<img width="1701" alt="Screenshot 2021-12-22 at 13 42 05" src="https://user-images.githubusercontent.com/3593343/147094477-67adbd02-8bea-427a-990e-b86a8659a9fe.png">

#### Related issue(s):

I'm adding the extra pattern from https://github.com/Automattic/themes/issues/5025 

Closes https://github.com/Automattic/themes/issues/5025

MaggieCabrera committed:
refactor blockbase and skatepark's footers into patterns

MaggieCabrera committed:
refactored videomaker's footer to a pattern

MaggieCabrera committed:
removed skatepark's pattern

MaggieCabrera committed:
refactored arbutus

MaggieCabrera committed:
renamed arbutus footer

MaggieCabrera committed:
quadrat, zoologist, geologist + variations

MaggieCabrera committed:
Kingsley and Mayland footers

MaggieCabrera committed:
Seedlet blocks footer

MaggieCabrera committed:
fix vertical footer pattern

MaggieCabrera committed:
added spacer to videomaker

MaggieCabrera committed:
block style for search and new footer pattern

kjellr commented:
> Should we change this text to something generic that works for all themes? Or maybe we should make this be the tagline block instead? @kjellr @beafialho I'd love your input on this

Yeah, the site tagline there is fine.

MaggieCabrera committed:
tagline on columns pattern

MaggieCabrera commented:
Rebased this one too. Changed the text on the skatepark pattern to the tagline.

@kjellr should we use the new pattern from #5025 on Geologist/Zoologist?

scruffian reviewed (approved):
LGTM
